pkg/cloud/externalconn/utils/connection_utils.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package utils
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/util/ioctx"
+	"github.com/cockroachdb/errors"
+)
+
+const markerFile = "crdb_external_storage_location"
+
+// CheckExternalStorageConnection writes a sentinel file, lists the file, and reads the file
+// back. This serves as a sanity check that the external connection represents
+// an ExternalStorage resource that can be connected and interacted with..
+func CheckExternalStorageConnection(
```

Comment:
I'm not sure how I feel about this check... There are few things that concern me:
1. Usually, this will happen under txn (create external connection has a txn), and I don't like making external mutations -- even if they are "invisible" while under txn.
2. The utility of this function is somewhat unclear since the fact that you can write file right now, does not imply the same in the future.
3. Consumers might be surprised to see marker file -- perhaps consumer only expects to see changefeed files...

Perhaps performing this validation after txn commits might be reasonable...

Reply:
We discussed this a little bit over in https://cockroachlabs.slack.com/archives/C03JCUUSCD6/p1660045972060119?thread_ts=1659997076.226179&cid=C03JCUUSCD6 but 

re: 1.
paraphrasing @dt, since create external connection is validate-then-insert the only way we're holding locks while validating is if a) you are in a txn that previously wrote (multi-stmt explicit txn) or b) you’re in a retry. We could ban a) and we did not particularly think b) is worth worrying about at the moment. 

re: 2.
While I agree with your point not doing _anything_ seems worse? A user creates an external connection and then goes through the motions of grants users `USAGE` of the conn, only for them to have a failed backup because our keys are wrong. Then they have to go back to dropping and recreating the connection (until we support alter) all over again.

re: 3.
I assumed it would be okay to docs this

Reply:
Or before the txn is opened? Is that possible?
Maybe we could delete the marker file too?

I like this test a lot - certainly it's not foolproof and 1000% futureproof, but nothing is, and my guess is this will eliminate a whole class of misconfigurations that our support teams will need to assist with

Reply:
>  Sounds like we're already doing this before txn is opened

Not exactly, the planners txn is open before we write it to the system table, but we do not perform the write until after the validation. I don't think there's an easy way to validate before the planners txn is opened. All this thinking out loud however made me realize we have a small bug around the creation of external connections and the txn we are using. Sending a patch separartely.